staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto

```diff
@@ -18,6 +18,15 @@ option (gogoproto.goproto_unrecognized_all) = false;
 // node resources consumed by pods and containers on the node
 service PodResourcesLister {
     rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+    rpc GetAllocatableResources(AllocatableResourcesRequest) returns (AllocatableResourcesResponse) {}
+}
+
+message AllocatableResourcesRequest {}
+
+// AllocatableResourcesResponses contains informations about all the devices known by the kubelet
+message AllocatableResourcesResponse {
+    repeated ContainerDevices devices = 1;
+    repeated int64 cpu_ids = 2;
```

Comment:
for the sake of completeness, can this should be made to include *all* container resources. There are use cases where, e.g., extended resources add value to get the big picture so be able to monitor full set of container resources is important.

Reply:
AFAIU this is an implementation detail and don't requires API (or KEP) changes, right?

Reply:
`AllocatableResourceResponse`  likely needs changes but it's in this PR so changes are possible.

To give an idea about the gap: I have a device plugin (`sgx.intel.com/enclave`) to allow containers to use devices nodes (`/dev/sgx_enclave`) and a separate extended resource to track how much SGX memory is needed (`sgx.intel.com/epc`).

With this API, I'm currently not able to monitor how much SGX memory each container is using. I can only see who are using the device node (which carries no resource information a such).

Reply:
Thanks for presenting this use case. 

This use case seems more a possible a gap in the List() endpoint rather in the `GetAllocatableResources` endpoint (which this PR concerns to).
In `GetAllocatableResources` we want to report how many resources does the node have, not the details of their usage (both in the sense of which pod uses them and the details of each resource usage, in this example the memory amount).
It still seems to me we need no changes in the `api,proto` spec. We can implement the reporting of the external resources, possibly in another PR.

In any case, should we need an `api.proto` extension, we will need to amend the KEP, which calls for more, separate PRs.

Reply:
KEP describing the original podresources API: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/606-compute-device-assignment
KEP extension this PR is (partially) implementing: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2043-pod-resource-concrete-assigments

Reply:
> In any case, should we need an api.proto extension, we will need to amend the KEP, which calls for more, separate PRs.

@frobware thanks for sharing the KEPs! I had missed the original one. I'll need to take a closer look which one (or is it both) need updates